third_party/gpus/rocm_configure.bzl

```diff
@@ -256,6 +256,36 @@ def _batch_files_exist(repository_ctx, libs_paths, bash_bin):
             all_paths.append(lib_path)
     return files_exist(repository_ctx, all_paths, bash_bin)
 
+
+def _soversion(repository_ctx, path, bash_bin = None):
+    """Returns the soversion of a given library.
+
+    Args:
+      repository_ctx: the repository_ctx
+      path: a path on the file system
+      bash_bin: path to the bash interpreter
+
+    Returns:
+      Parsed soversion string form the SONAME dtag of the library
+    """
+    if bash_bin == None:
+        bash_bin = get_bash_bin(repository_ctx)
+
+    exec_result = execute(repository_ctx, [bash_bin, "-c", "readelf --dynamic \"%s\"" % path])
+
+    if exec_result.return_code:
+        auto_configure_fail("Failed to run readelf to find soversion: %s" % err_out(exec_result))
+
+    soversion = ""
+    for row in exec_result.stdout.strip().split("\n"):
+        match = row.find("SONAME")
+        if match >= 0:
+           match = row.find(".so.", match)
+           if match >= 0:
+              soversion = row[match + 4: -1]
+              break
+    return soversion
```

Comment:
:) since we cannot use regrex, let's not assume version is always at the end of line?

how about this?

```
soversion = ""
for row in exec_result.stdout.strip().split("\n"):
    if "SONAME" in row:
        start = row.find("[")
        end = row.find("]", start)
        if start != -1 and end != -1:
            soname = row[start + 1:end]
            so_index = soname.find(".so.")
            if so_index != -1:
                soversion = soname[so_index + 4:]
                break
```